test/integration/proxy_req_test.js

```diff
@@ -287,3 +288,73 @@ test('removes some timeouts, not others', function t(assert) {
         });
     });
 });
+
+test('overrides /proxy/req endpoint', function t(assert) {
+    assert.plan(3);
+
+    var endpoint = 'FIND ME A THING';
+    var things = [
+        'thing1',
+        'thing2',
+        'thing3'
+    ];
+
+    var cluster = allocCluster(function onReady() {
+        cluster.two.on('request', function onRequest() {
+            assert.fail('did not bypass request proxy handler');
+        });
+
+        cluster.two.channel.register(endpoint, function handler(arg1, arg2, hostInfo, cb) {
+            assert.equal(arg1.toString(), head, 'arg1 is raw head');
+            assert.equal(arg2.toString(), '{"hello":true}', 'arg2 is raw body');
+
+            cb(null, arg1, things);
+        });
+
+        var request = cluster.request({
+            key: cluster.keys.two,
+            host: 'one',
+            endpoint: endpoint,
+            json: { hello: true },
+            maxRetries: 0
+        }, function onRequest(err, resp) {
+            assert.deepEqual(resp.body, things, 'responds with body');
+
+            cluster.destroy();
+            assert.end();
+        });
+
+        var head = serializeHead(request, {
```

Comment:
can't do that because of `request` used as first param. i originally assigned `head` within the `register` call, but that seemed a little too late.